app/tests/api/logging/test_formatters.py

```diff
@@ -0,0 +1,44 @@
+import json
+import logging
+import re
+
+import pytest
+
+import api.logging.formatters as formatters
+
+
+def test_json_formatter(capsys: pytest.CaptureFixture):
+    logger = logging.getLogger("test_json_formatter")
+    console_handler = logging.StreamHandler()
+    console_handler.setFormatter(formatters.JsonFormatter())
+    logger.addHandler(console_handler)
+
+    logger.warning("hello", extra={"foo": "bar"})
+
+    json_record = json.loads(capsys.readouterr().err)
+    expected = {
+        "name": "test_json_formatter",
+        "msg": "hello",
+        "levelname": "WARNING",
+        "levelno": 30,
+        "filename": "test_formatters.py",
+        "module": "test_formatters",
+        "funcName": "test_json_formatter",
+        "foo": "bar",
+    }
+    assert json_record | expected == json_record
```

Comment:
What's the advantage over doing `json_record == expected`?

Reply:
it's because the the log record contains a lot of other data that i don't really care about, like line number, thread number etc. if i change it to json_record == expected i get the following diff

<img width="1529" alt="image" src="https://user-images.githubusercontent.com/447859/214140113-5265f90f-dbef-4a99-8313-d3a251f7f255.png">

this is essentially the same as `_assert_dict_contains` in `test_flask_logger.py`

should i just add a comment here explaining it? or maybe create a `_assert_dict_contains` test utility?

Reply:
I moved it to a util `tests.lib.assertions`